tests/robot-tests/tests/libs/slack.py

```diff
@@ -0,0 +1,96 @@
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+import shutil
+import json
+import requests
+from bs4 import BeautifulSoup
+
+PATH = f'{os.getcwd()}{os.sep}test-results'
+
+
+def _generate_slack_attachments():
+    failed_tests = dict()
+    passed_tests = dict()
+    skipped_tests = dict()
+
+    report = open(f'{PATH}{os.sep}output.xml', 'rb')
+    contents = report.read()
+
+    soup = BeautifulSoup(contents, 'xml')
+    test = soup.find('total').find('stat')
+
+    failed_tests = test['fail']
+    passed_tests = test['pass']
+    skipped_tests = test['skip']
+
+    return [
+        {
+            "pretext": "All results",
+            "color": "danger" if failed_tests or skipped_tests is not None or not "" else "good",
+            "mrkdwn_in": [
+                "pretext"
+            ],
+            "fields": [
+                {
+                    "title": "Total test cases",
+                    "value": int(skipped_tests) + int(passed_tests) + int(failed_tests)
+                },
+                {
+                    "title": "Passed tests",
+                    "value": int(passed_tests)
+                },                
+                {
+                    "title": "Failed tests",
+                    "value": int(failed_tests) + int(skipped_tests) if failed_tests or skipped_tests is not None or not "" else "0"
+                },
+                {
+                    "title": "Results",
+                    "value": "Failed" if failed_tests or skipped_tests is not None or not "" else "Passed"
+                },
+               
+            ]
+        }
+    ]
+
+
+def send_slack_report():
+    attachments = _generate_slack_attachments()
+    data = {"attachments": attachments}
+
+    webhook_url = os.getenv('SLACK_TEST_REPORT_WEBHOOK_URL')
+    slack_bot_token = os.getenv('SLACK_BOT_TOKEN')
+
+    assert webhook_url is not None or not "", print("SLACK_TEST_REPORT_WEBHOOK_URL env variable needs to bet set")
+    assert slack_bot_token is not None or not "", print("SLACK_BOT_TOKEN env variable needs to bet set")
+
+    requests.sessions.HTTPAdapter(
+            pool_connections=50,
+            pool_maxsize=50,
+            max_retries=3
+        )
+    session = requests.Session()
```

Comment:
I'm guessing this is the connection pooling stuff? What benefit does it provide, given that we run this right at the end of the test run? 

My assumption would be that there'd be no benefit as we're only making a single request and the test runner is about to complete anyway. Wouldn't the work to set up the connection pool would just be thrown away?

Reply:
Yep, that's a good point. I've refactored this to use `requests.post`